policies.go

```diff
@@ -145,8 +144,9 @@ var ErrUnknownRetryType = errors.New("unknown retry type returned by retry polic
 // See SimpleRetryPolicy as an example of implementing and using a RetryPolicy
 // interface.
 type RetryPolicy interface {
-	Attempt(RetryableQuery) bool
+	Attempt(ExecutableQuery, Iter) bool
 	GetRetryType(error) RetryType
+	GetRetryRequest(ExecutableQuery, Iter, error) ExecutableQuery
```

Comment:
ehh these changes to `RetryPolicy` make me a bit concerned... I think there's probably a more sophisticated way to do this without breaking custom retry policies.
Maybe we can explore changes to `RetryableQuery` instead and maybe making `Query` and `Batch` not implement `RetryableQuery` and either making `Iter` implement this interface or creating a new private type that implements this interface and this new private object would be a field of `Iter`

Reply:
Or we just copy the `Query`/`Batch` objects when we start a request execution and keep `Query` and `Batch` as implementations of `RetryableQuery` in case a retry policy wants to access the `Query` and `Batch` information that "generated" that request